internal/output/prefixed.go

```diff
@@ -5,22 +5,23 @@ import (
 	"fmt"
 	"io"
 	"strings"
+	"sync"
 
 	"github.com/go-task/task/v3/internal/logger"
 	"github.com/go-task/task/v3/internal/templater"
 )
 
 type Prefixed struct {
 	logger  *logger.Logger
-	seen    map[string]uint
+	seen    *sync.Map
```

Comment:
Can we simply use a mutex for protecting the `seen` map? It's actively used in this project: [1](https://github.com/go-task/task/blob/c28eb204fb404ea141e5b68c2765d7386a166780/internal/compiler/compiler.go#L32), [2](https://github.com/go-task/task/blob/c28eb204fb404ea141e5b68c2765d7386a166780/task.go#L84).

```suggestion
	muSeen  sync.Mutex
	seen    map[string]uint
```

Additionally, `sync.Map` has drawbacks and rare use cases. See https://victoriametrics.com/blog/go-sync-map/

Reply:
From the go docs of `sync.Map`:

> The Map type is optimized for two common use cases: (1) when the entry for a given
> key is only ever written once but read many times, as in caches that only grow,
> or (2) ...

I think that applies here, because potentially a lot of lines get written and only relatively few prefixes will be used.

I think speed is important here since task execution blocks waiting for output to be written. But I have no strong opinion about either way, and we would have to benchmark to check which is actually faster.

One drawback of `sync.Map` over a mutex is that `sync.Map` is not type safe, but I think its usage is easy to oversee in this case.